can/typechecking.py

```diff
@@ -18,3 +21,8 @@
 
 # Used for the Abstract Base Class
 Channel = typing.Union[int, str]
+
+# Used by the IO module
+FileLike = typing.IO[typing.Any]
+PathLike = typing.Union[str, "os.PathLike[str]"]
```

Comment:
Technically, I believe `open` supports `str`, `bytes`, something that implements the `os.PathLike` interface, or an `int` corresponding to the file descriptor of the file. Are we just planning on ignoring `bytes` and `int` representations?

Reply:
Yeah, that was roughly my thought as well. But I now think we should be able to support `byte` and `int` as well, since `PathLike` needs to implement `__fs_path__` (or something that) anyways, so we will always have a path to compare with.

Reply:
You were right, only `str` works.